deepmd/utils/transfer_to_ascend.py

```diff
@@ -0,0 +1,147 @@
+                "Note that the input transfering script must contain the correct path to the training data." % input
+            ) from e
+        elif not os.path.exists(training_script):
+            raise RuntimeError(
+                "The input training script %s (%s) does not exist! Please check the path of the training script. " % (input, os.path.abspath(input))
+            ) from e
+        else:
+            log.info("stage 0: compute the min_nbor_dist")
+            jdata = j_loader(training_script)
+            jdata = update_deepmd_input(jdata)
+            t_min_nbor_dist = get_min_nbor_dist(jdata, get_rcut(jdata))
+
+    _check_transfer_model_type(input)
+
+    tf.constant(t_min_nbor_dist,
+        name = 'train_attr/min_nbor_dist',
+        dtype = GLOBAL_ENER_FLOAT_PRECISION)
+    jdata["model"]["descriptor"]["precision"] = "float16"
+    jdata["model"]["fitting_net"]["precision"] = "float16"
+    jdata["model"]["transfered_from_model"] = True
+    jdata["training"]["save_ckpt"] = os.path.join("model-transfer", "model.ckpt")
+    jdata = update_deepmd_input(jdata)
+    jdata = normalize(jdata)
+
+    # check the descriptor info of the input file
+    # move to the specific Descriptor class
+
+    # stage 1: training or refining the model with tabulation
+    log.info("\n\n")
+    log.info("stage 1: generate the mix-precision model")
+    control_file = "ascend-transfer.json"
+    with open(control_file, "w") as fp:
+        json.dump(jdata, fp, indent=4)
+    try:
+        train(
+            INPUT=control_file,
+            init_model=None,
+            restart=None,
+            init_frz_model=input,
+            output=control_file,
+            mpi_log=mpi_log,
+            log_level=log_level,
+            log_path=log_path,
+            is_compress=False,
+            skip_neighbor_stat=True,
+        )
+    except GraphTooLargeError as e:
+        raise RuntimeError(
+            "The uniform step size of the tabulation's first table is %f, " 
+            "which is too small. This leads to a very large graph size, "
+            "exceeding protobuf's limitation (2 GB). You should try to "
+            "increase the step size." % step
+        ) from e
+
+    # reset the graph, otherwise the size limitation will be only 2 GB / 2 = 1 GB
+    tf.reset_default_graph()
+
+    # stage 2: freeze the mix-precision model
+    log.info("\n\n")
+    log.info("stage 2: freeze the mix-precision model")
+    try:
+        freeze(checkpoint_folder=checkpoint_folder, output=output, node_names=None)
+    except GraphTooLargeError as e:
+        raise RuntimeError(
+            "The uniform step size of the tabulation's first table is %f, " 
+            "which is too small. This leads to a very large graph size, "
+            "exceeding protobuf's limitation (2 GB). You should try to "
+            "increase the step size." % step
```

Comment:
`step` is not defined.

Reply:
Thanks for the comment, we have modified the error explanation.